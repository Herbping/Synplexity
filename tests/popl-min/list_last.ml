type nat =
  | Z
  | S of nat

type list =
  | Nil
  | Cons of nat * list

type natopt =
  | None
  | Some of nat

let rec list_last : list -> natopt |>
 /\([]              -> None,
    \/([2; 1], [1]) -> Some(1),
    [2]             -> Some(2)) = ?


