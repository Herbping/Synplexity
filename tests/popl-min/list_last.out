(* Generated Thu Oct  8 20:38:38 2015 *)
fix f1 (l1:list) : natopt ->
  match l1 with
  | Nil u1 -> None
  | Cons t1 -> match f1 (#2 t1) with
    | None u1 -> Some (#1 t1)
    | Some n1 -> Some n1
